src/Microsoft.AspNetCore.DataProtection/EphemeralDataProtectionProvider.cs

```diff
@@ -48,10 +48,10 @@ public EphemeralDataProtectionProvider(IServiceProvider services)
                 keyringProvider = new EphemeralKeyRing<ManagedAuthenticatedEncryptionSettings>();
             }
 
-            var logger = services.GetLogger<EphemeralDataProtectionProvider>();
+            var logger = loggerFactory?.CreateLogger<EphemeralDataProtectionProvider>();
             logger?.UsingEphemeralDataProtectionProvider();
```

Comment:
Yes,  this call can be moved outside this class but it still needs the `ILoggerFactory` which is passed in here https://github.com/aspnet/DataProtection/pull/136/files#diff-46b998be1bdad1ab2c77fb0e3b161b7aR54. I am still checking if we can switch to using some kind of context class.